gunrock/graph/dynamic_graph/dynamic_graph_weighted.cuh

```diff
@@ -0,0 +1,151 @@
+// ----------------------------------------------------------------------------
+// Gunrock -- Fast and Efficient GPU Graph Library
+// ----------------------------------------------------------------------------
+// This source code is distributed under the terms of LICENSE.TXT
+// in the root directory of this source distribution.
+// ----------------------------------------------------------------------------
+
+/**
+ * @file
+ * dynamic_graph_weighted.cuh
+ *
+ * @brief DYN (Dynamic) Graph Data Structure
+ */
+#pragma once
+
+#include <gunrock/graph/dynamic_graph/dynamic_graph.cuh>
+#include <gunrock/graph/dynamic_graph/dynamic_graph_base.cuh>
+
+namespace gunrock {
+namespace graph {
+
+/**
+ * @brief Weighted dynamic graph data structure which uses
+ * a per-vertex data structure based on the graph flags.
+ *
+ * @tparam VertexT Vertex identifier type.
+ * @tparam SizeT Graph size type.
+ * @tparam ValueT Associated value type.
+ * @tparam GraphFlag graph flag
+ */
+template<
+    typename VertexT,
+    typename SizeT,
+    typename ValueT,
+    GraphFlag FLAG,
+    unsigned int cudaHostRegisterFlag>
+struct Dyn<VertexT, SizeT, ValueT, FLAG, cudaHostRegisterFlag, true, true> : public DynamicGraphBase<VertexT, SizeT, ValueT, FLAG> {
```

Comment:
`struct` defaults to public inheritance.

Reply:
Fixed https://github.com/gunrock/gunrock/pull/713/commits/127da41c090f97105b46b91c36907068a040410c